task1/Fridge.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace task1
+{
+    public delegate void FridgeStateEventHandler(object sender, FridgeHandlerEventArgs args);
+
+    public class Fridge
+    {
+        public State FridgeState { get; set; }
+
+        public Door FridgeDoor { get; set; }
+
+        public string ActionMessage { get; set; }
+
+        public event FridgeStateEventHandler FridgeStateChangedEvent;
+
+        public Fridge()
+        {
+            this.ActionMessage = string.Empty;
+        }
+
+        public void GetFridgeState()
+        {
+            Console.WriteLine($"Fridge is {this.FridgeState} and {this.FridgeDoor} is open");
+        }
+
+        public void OpenDoor()
+        {
+            Console.WriteLine("What door to open R or F?");
+            string door = Console.ReadLine();
+            if (door == "R")
+            {
+                this.FridgeDoor = Door.RegularCamera;
+                this.ActionMessage = "Door of regular camera has been opened\n";
+            }
+            if (door == "F")
```

Comment:
Please use 'else if' statement to avoid excessive checks. Door can not store "R" and "F" values at the same time.